packages/preprocess/src/process-queries.cjs

```diff
@@ -0,0 +1,164 @@
+        
+        let props;
+        export { props as data }; // little hack to make the data name not overlap
+        let { data = {}, customFormattingSettings } = props;
+        $: ({ data = {}, customFormattingSettings } = props);
+
+        $routeHash = '${routeH}';
+
+        $: data, Object.keys(data).length > 0 ? pageHasQueries.set(true) : pageHasQueries.set(false);
+
+        setContext(CUSTOM_FORMATTING_SETTINGS_CONTEXT_KEY, {
+            getCustomFormats: () => {
+                return customFormattingSettings.customFormats || [];
+            }
+        });
+        
+        const applyEvidenceTypes = function(data) {
+
+            let includedQueries = data.evidencemeta?.queries
+
+            if(includedQueries) {
+                // iterate through each query 
+                for(let i = 0; i < includedQueries.length; i++) {
+                    // for each of the query objects in data
+                    let query = data[includedQueries[i].id]
+                    let colTypes = data.evidencemeta?.queries[i].columnTypes
+                    // iterate through each row in the query
+                    for(let j = 0; j < query.length; j++) {
+                        // for each row in the query
+                        if(colTypes) {
+                            // include column types in the row object as a non enumerable property
+                            Object.defineProperty(query[j], '_evidenceColumnTypes', {
+                                enumerable: false,
+                                value: colTypes,
+                            });
+                        }
+                    }
+                }
+            }
+    
+        }
+    
+        beforeUpdate(() => {
+            applyEvidenceTypes(data)
+        })
+
+        ${queryDeclarations}
+        `
+
+    return defaultProps
+}
+// Unified parser step to ignore indented code blocks. 
+// Adapted from the mdsvex source, here: https://github.com/pngwn/MDsveX/blob/master/packages/mdsvex/src/parsers/index.ts
+// Discussion & background here:  https://github.com/evidence-dev/evidence/issues/286
+const ignoreIndentedCode = function() {
+	const Parser = this.Parser;
+	const block_tokenizers = Parser.prototype.blockTokenizers;
+	block_tokenizers.indentedCode = () => true;
+}
+
+const getQueryIds = function(content){
+    let queryIds = [];  
+    let tree = unified()
+        .use(parse)
+        .use(ignoreIndentedCode)
+        .parse(content)   
+
+    visit(tree, 'code', function(node) {
+        let id = node.lang ?? 'untitled'
+         // Prevent "real" code blocks from being interpreted as queries
+         if (!getPrismLangs().has(id.toLowerCase())){
+             queryIds.push(id)
+         }
+    });
+    return queryIds;
+}
+
+/**
+ * @type {(componentDevelopmentMode: boolean) => import("svelte-preprocess/dist/types").PreprocessorGroup}
+ */
+const processQueries = (componentDevelopmentMode) => {
+    let queryIdsByFile = {}
```

Comment:
This `queryIdsByFile` was bit of a strange "workaround" as we needed remember state between different pre-process phases (without having to load things twice from what I recall).   I think this works neatly with way you refactored this, under the assumption that the order of the array returned by `evidencePreprocess` doesn't matter.   I think it doesn't but I am not sure.